src/ripple/protocol/impl/InnerObjectFormats.cpp

```diff
@@ -59,6 +61,58 @@ InnerObjectFormats::InnerObjectFormats()
             {sfNFTokenID, soeREQUIRED},
             {sfURI, soeOPTIONAL},
         });
+
+    add(sfXChainClaimAttestationBatchElement.jsonName.c_str(),
+        sfXChainClaimAttestationBatchElement.getCode(),
+        {
+            {sfAttestationSignerAccount, soeREQUIRED},
+            {sfPublicKey, soeREQUIRED},
```

Comment:
First, I've exercised the changes that support multisigners using a regular key.  It's working great for me.  Thanks for doing that.

Here's tweak to consider.  You have an `AttestationSignerAccount` field.  That's a great name.  I really understand what that field is doing.  But the public key that goes with that field is just called `PublicKey`.  When I look at the fields of an `XChainAddClaimAttestation` transaction it's not obvious what that `PublicKey` is for, or even what it's associated with.

I suggest you rename the `PublicKey` field to `AttestationSignerPublicKey`.  That will make the documentation for the transaction a lot clearer.  And there's a lot going on in that transaction, so it's easy to get confused.

Possibly the `Signature` field should get the same treatment and be renamed to `AttestationSignature`?  There's a similar kind of ambiguity between the `Signature` and `TxnSignature` fields.

Just a suggestion.

A different approach, if it turns out that `STObjects` can contain `STObjects`, would be to gather up all of the attestation stuff in an object.  Like this:
```
    "Attestation" : {
        "Account" : "...",
        "SigningPubKey" : "...",
        "TxnSignature" : "..."
    },
```
If that were done then Attestations and multi signatures would have very similar structures.  But that would be a bigger change.